### ISSUE TITLE
Build broken after minor 3.6.0

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/vscode-languageserver-node#317]
I am getting the following errors with `vscode-languageserver@3.6.0` that were not happening in 3.5:
```
node_modules/vscode-languageserver/lib/configuration.proposed.d.ts(6,37): error TS2694: Namespace '"/Users/felix/git/sourcegraph/javascript-typescript-langserver/node_modules/vscode-languageserver-protocol/lib/main".Proposed' has no exported member 'ConfigurationItem'.
node_modules/vscode-languageserver/lib/configuration.proposed.d.ts(7,38): error TS2694: Namespace '"/Users/felix/git/sourcegraph/javascript-typescript-langserver/node_modules/vscode-languageserver-protocol/lib/main".Proposed' has no exported member 'ConfigurationItem'.
node_modules/vscode-languageserver/lib/workspaceFolders.proposed.d.ts(4,46): error TS2694: Namespace '"/Users/felix/git/sourcegraph/javascript-typescript-langserver/node_modules/vscode-languageserver-protocol/lib/main".Proposed' has no exported member 'WorkspaceFolder'.
node_modules/vscode-languageserver/lib/workspaceFolders.proposed.d.ts(5,49): error TS2694: Namespace '"/Users/felix/git/sourcegraph/javascript-typescript-langserver/node_modules/vscode-languageserver-protocol/lib/main".Proposed' has no exported member 'WorkspaceFoldersChangeEvent'.
src/test/typescript-service-helpers.ts(92,25): error TS2345: Argument of type '{ processId: number; rootUri: string; capabilities: ClientCapabilities; }' is not assignable to parameter of type 'InitializeParams'.
  Property 'workspaceFolders' is missing in type '{ processId: number; rootUri: string; capabilities: ClientCapabilities; }'.
```

**Comment by felixfbecker** [comment]
https://travis-ci.org/felixfbecker/vscode-php-intellisense/jobs/349595813

**Comment by yungez** [comment]
met same error. `vscode-languageclient@3.5.0`, `vscode-languageserver-protocol@3.6.0`

node_modules/vscode-languageclient/lib/configuration.proposed.d.ts(5,34): error TS2694: Namespace '"C:/projects/vscode-ansible/node_modules/vscode-languageserver-protocol/lib/main".Proposed' has no exported member 'ConfigurationRequest'.
572node_modules/vscode-languageclient/lib/workspaceFolders.proposed.d.ts(6,37): error TS2694: Namespace '"C:/projects/vscode-ansible/node_modules/vscode-languageserver-protocol/lib/main".Proposed' has no exported member 'WorkspaceFoldersRequest'.

**Comment by dbaeumer** [comment]
The problem is that the 3.5.0 client had a protocol dependency ^3.5.0 which when you install the client it gets 3.6.0 which has the proposed API removed and made official. Same is true for server. I published a 3.5.1 version of the client and the server which version locks on 3.5.1 of the protocol which version locks on json-rpc and types.

Please note that proposed protocol always stated that if might break existing clients even if the versioning doesn't follow semver. However a normal install of the package should still do the right things. Please let me know if your builds are now passing.

**Comment by dbaeumer** [comment]
Closing the issue. Please ping if this doesn't solve it.
